flink-runtime/src/main/java/org/apache/flink/runtime/asyncprocessing/declare/state/StateWithDeclaredNamespace.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.asyncprocessing.declare.state;
+
+import org.apache.flink.annotation.Internal;
+import org.apache.flink.api.common.state.v2.State;
+import org.apache.flink.runtime.asyncprocessing.declare.DeclaredVariable;
+import org.apache.flink.runtime.state.v2.internal.InternalAggregatingState;
+import org.apache.flink.runtime.state.v2.internal.InternalListState;
+import org.apache.flink.runtime.state.v2.internal.InternalMapState;
+import org.apache.flink.runtime.state.v2.internal.InternalPartitionedState;
+import org.apache.flink.runtime.state.v2.internal.InternalReducingState;
+import org.apache.flink.runtime.state.v2.internal.InternalValueState;
+
+/**
+ * State wrapped by declared namespace. This is useful for window operator, where we need to
+ * frequently setup current namespace in each lambda. This wrap class help use to invoke {@code
+ * setCurrentNamespace} when we are going to call async state API.
+ */
+@Internal
+public class StateWithDeclaredNamespace<N> {
+    private final InternalPartitionedState<N> state;
+    private final DeclaredVariable<N> declaredNamespace;
+
+    public StateWithDeclaredNamespace(State state, DeclaredVariable<N> declaredNamespace) {
+        if (state instanceof InternalPartitionedState) {
+            this.state = (InternalPartitionedState<N>) state;
+        } else {
+            this.state = null;
+        }
+        this.declaredNamespace = declaredNamespace;
+    }
+
+    protected void wrapNamespace() {
```

Comment:
A little strange here... How about naming it `resetNamespace`

Reply:
resolved